### ISSUE TITLE
Messages are not rendered when the system uses a new timezone

### ISSUE THREAD

**Issue by ForNeVeR** [nylas/N1#1964]
![image](https://cloud.githubusercontent.com/assets/92793/14532457/5ec591de-028b-11e6-8b4b-651b7a36bf4d.png)

After update to 0.4.25 messages are no longer visible in the main pane. The same problem has been reproduced on both of my N1 installations (Windows 8.1 and Windows 10).

I can see that N1 starts new process with the following command every (or almost every) time I try to view a message:

```
nylas.exe --eval require('C:\\Users\\ForNeVeR\\AppData\\Local\\nylas\\app-0.4.25\\resources\\app.asar\\src\\compile-cache.js').setCacheDirectory('C:\\Users\\ForNeVeR\\.nylas\\compile-cache');
require('C:\\Users\\ForNeVeR\\AppData\\Local\\nylas\\app-0.4.25\\resources\\app.asar\\src\\task-bootstrap.js');
```

These processes never exit and tend to just stay in memory without any CPU consumption.

I've tried to start N1 in debug mode (through menu -> Developer -> Run with Debug flags), and it fixed the problem on Windows 8.1 (so I can view my mail in debug mode now). On Windows 10 I got the following error in the debug console: https://gist.github.com/ForNeVeR/85de4f48e9dc3ab1ffacb34c3b440131

##### What operating system are you using?
Windows 8.1 x64, Windows 10 x64.

##### What version of N1 are you using?
Both were updated to 0.4.25.

##### Do you have any third-party plugins installed?
No.

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
Don't think so, but possible. I am using the same set of accounts on both of my N1 installations: one Yandex Mail for Domains and three GMail accounts.

##### Is the issue reproducible with a particular attachment, message, signature, etc?
It reproduces for every message in my inbox. I cannot view any of them.

**Comment by bengotow** [comment]
Hmmm that's really strange. I think the task is spawning to ship N1's logs to our log collection service. I'll check there and see if anything has come in. Also, I imagine the messages don't appear if you click this button?

![image](https://cloud.githubusercontent.com/assets/1037212/14540374/ae194a8e-0252-11e6-8ede-445b3c6b6f7a.png)

**Comment by ForNeVeR** [comment]
Unfortunately, I cannot see this button no more on any of mu current (or deleted) emails, so I cannot check it, sorry. I'll try to research the issue deeper tomorrow. It doesn't cause too much trouble for me because I can read messages using alternate email clients, but I want my N1 back online :)

**Comment by ForNeVeR** [comment]
Well, I've debugged that a bit, and found, that the problem lurks in [this line](https://github.com/nylas/N1/blob/master/src/flux/models/utils.coffee#L71) of `utils.coffee`. `Utils.timeZone` is `undefined` there, and that causes problems.

`Utils.timeZone` itself is initialized from `Intl.DateTimeFormat().resolvedOptions().timeZone`, which is undefined too in my environment. It's linked with the recent timezone update in my country. Currently I use a timezone "UTC +7 Barnaul, Gorno-Altaysk", that was added in the latest Windows update.

I can confirm that the new timezone causes this bug. I've switched my system to use one of the old time zones, and the issue was fixed. It looks like some library vendor (or Electron itself, or whatever) still haven't updated their timezone database.

So, that's definitely a timezone issue. What actions should I take next? I guess it's a library issue, so I should raise it in some external bug tracker?

**Comment by frigaut** [comment]
Same problem here. Running 0.4.25-a22631a on a linux box (arch).
Changing my timezone indeed fixes the issue.

**Comment by ForNeVeR** [comment]
Unfortunately, still reproducible on 0.4.32 :(

**Comment by bengotow** [comment]
Hey! Was this resolved by https://github.com/nylas/N1/pull/2250?

**Comment by bengotow** [comment]
Hey folks—I can confirm that this was fixed by the PR above (#2250), and we've also bumped Moment in 17d9c3d7175392538345e32e54ea871987f85c2d so it includes the new time zones. Stay tuned.

**Comment by bengotow** [comment]
Hey! Thanks for reporting this—we shipped `0.4.45` this afternoon which should resolve this issue. It should ship via the autoupdater in the next 24 hours, or you can download it directly from the Releases page. I'm going to ahead and close this—if you continue seeing the issue, feel free to comment and I'll re-open it!
